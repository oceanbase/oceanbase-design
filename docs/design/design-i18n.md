---
title: 国际化
order: 4
group: 基础组件
---

`@oceanbase/design` 默认语言是中文，如果要支持其他语言，需要手动开启国际化。

## 开启国际化

可通过 ConfigProvider 全局配置开启国际化，`ConfigProvider` 使用 `React` 的 `context` 特性，只需在应用外围包裹一次即可全局生效。

```jsx | pure
import { ConfigProvider } from '@oceanbase/design';
import zhCN from '@oceanbase/design/es/locale/zh-CN';

return (
  <ConfigProvider locale={zhCN}>
    {...}
  </ConfigProvider>
);
```

## 使用示例

<!-- prettier-ignore -->
<code src="../../packages/design/src/locale/demo/basic.tsx"></code>

## 目前支持语言

目前支持以下语言：

| 语言     | 文件名 |
| -------- | ------ |
| 简体中文 | zh-CN  |
| 繁体中文 | zh-TW  |
| 英文     | en-US  |

如当前国际化语言有问题或者需要支持其他语言，欢迎反馈给我们。
