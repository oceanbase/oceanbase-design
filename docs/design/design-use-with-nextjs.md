---
title: 在 Next.js 中使用
order: 3
group: 基础组件
---

- 为了和 `@oceanbase/design` 搭配使用，需要安装 `file-loader` 依赖，并修改 Next.js 配置：

```bash
npm install file-loader --save-dev
```

```ts
const nextConfig = {
  // to transpile some special packages to es5
  transpilePackages: ['@oceanbase', 'query-string', '@ant-design', '@ant-design/cssinjs'],
  webpack: config => {
    // to handle @oceanbase/design built-in font files
    config.module.rules.push({
      test: /\.(woff|woff2|ttf?g)$/i,
      use: ['file-loader'],
    });
    return config;
  },
};

export default nextConfig;
```

- 完整配置和可运行的示例，可以参考我们提供的 [Next.js + @oceanbase/design 项目模板](https://stackblitz.com/edit/nextjs-oceanbase-design)。

<iframe src="https://stackblitz.com/edit/nextjs-oceanbase-design?embed=1&file=app%2Fpage.tsx"
     style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
     title="@oceanbase/ui reproduction template"
     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
   ></iframe>
