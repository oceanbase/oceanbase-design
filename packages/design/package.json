{
  "name": "@oceanbase/design",
  "version": "0.4.10",
  "description": "The Design System of OceanBase",
  "keywords": [
    "oceanbase",
    "design",
    "ui",
    "react",
    "component"
  ],
  "homepage": "https://github.com/oceanbase/oceanbase-design/packages/design",
  "repository": {
    "type": "git",
    "url": "git@github.com:oceanbase/oceanbase-design.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "main": "lib/index.js",
  "unpkg": "dist/design.min.js",
  "module": "es/index.js",
  "typings": "es/index.d.ts",
  "files": [
    "es",
    "lib",
    "dist",
    "package.json",
    "README.md",
    "LICENSE"
  ],
  "sideEffects": [
    "**/*.less",
    "**/*.css",
    "es/index.js"
  ],
  "scripts": {
    "build": "father build && cp src/style/reset.css dist/"
  },
  "dependencies": {
    "@ant-design/cssinjs": "^1.23.0",
    "@ctrl/tinycolor": "^4.1.0",
    "@inline-svg-unique-id/react": "^1.2.3",
    "@oceanbase/aliyun-theme": "^0.1.6",
    "@oceanbase/icons": "workspace:^",
    "@oceanbase/util": "workspace:^",
    "ahooks": "^2.10.14",
    "antd": "^5.24.7",
    "classnames": "^2.5.1",
    "lodash": "^4.17.21",
    "lottie-web": "^5.12.2",
    "prop-types": "^15.8.1",
    "rc-util": "^5.44.4"
  },
  "devDependencies": {
    "@babel/cli": "^7.27.0",
    "@babel/preset-env": "^7.26.9",
    "antd-token-previewer": "^2.0.8",
    "rc-select": "^14.16.6",
    "rc-slider": "^11.1.8",
    "rc-table": "^7.50.4",
    "rc-tree-select": "^5.27.0"
  },
  "peerDependencies": {
    "react": ">=16.9.0",
    "react-dom": ">=16.9.0"
  }
}
