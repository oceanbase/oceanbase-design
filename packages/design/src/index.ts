import './global.css';

export * from 'antd';
export { version } from '../package.json';

export { default as Alert } from './alert';
export type { AlertProps } from './alert';

export { default as Button } from './button';
export type { ButtonProps } from './button';

export { default as Space } from './space';
export type { SpaceProps, SpaceSize } from './space';

export { Col, Row } from './grid';
export type { ColProps, RowProps, ColSize } from './grid';

export { default as Typography } from './typography';
export type { TypographyProps } from './typography';

export { default as Card } from './card';
export type { CardProps } from './card';

export { default as ConfigProvider } from './config-provider';
export type { ConfigProviderProps, ConfigConsumerProps, ThemeConfig } from './config-provider';

export { default as Descriptions } from './descriptions';
export type {
  DescriptionsProps,
  DescriptionsItemType,
  DescriptionsItemProps,
} from './descriptions';

export { default as Empty } from './empty';
export type { EmptyProps } from './empty';

export { default as Form } from './form';
export type { FormProps, FormItemProps } from './form';

export { default as Modal } from './modal';
export type { ModalProps, ModalFuncProps, ModalProgressProps } from './modal';

export { default as Drawer } from './drawer';
export type { DrawerProps } from './drawer';

export { message, notification, token } from './static-function';

export { default as List } from './list';
export type { ListProps } from './list';

export { default as Table } from './table';
export type { TableProps } from './table';

export { default as Tabs } from './tabs';
export type { TabsProps } from './tabs';

export { default as Tag } from './tag';
export type { TagProps } from './tag';

export { default as Input } from './input';
export type { InputProps } from './input';

export { default as InputNumber } from './input-number';
export type { InputNumberProps } from './input-number';

export { default as Radio } from './radio';
export type { RadioProps } from './radio';

export { default as Checkbox } from './checkbox';
export type { CheckboxProps } from './checkbox';

export { default as Switch } from './switch';
export type { SwitchProps } from './switch';

export { default as Select } from './select';
export type { SelectProps } from './select';

export { default as TreeSelect } from './tree-select';
export type { TreeSelectProps } from './tree-select';

export { default as Slider } from './slider';
export type { SliderSingleProps, SliderRangeProps } from './slider';

export { default as Tooltip } from './tooltip';
export type { TooltipProps } from './tooltip';

export { default as Segmented } from './segmented';
export type { SegmentedProps } from './segmented';

export { default as Breadcrumb } from './breadcrumb';
export type { BreadcrumbProps, BreadcrumbItemProps } from './breadcrumb';

export { default as Result } from './result';
export type { ResultProps, ResultType, ResultStatusType } from './result';

export { default as Spin } from './spin';
export type { SpinProps } from './spin';

export { default as Badge } from './badge';
export type { BadgeProps } from './badge';

export { default as Lottie } from './lottie';
export type { LottieProps, LottieRef } from './lottie';

export { default as theme } from './theme';

import theme from './theme';

const { useToken } = theme;
export { useToken };

export type { PresetStatusColorType } from 'antd/es/_util/colors';
export { genLargeStyle } from './_util/genStyle';
