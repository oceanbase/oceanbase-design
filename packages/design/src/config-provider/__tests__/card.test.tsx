import React, { useContext } from 'react';
import { render } from '@testing-library/react';
import { Card, ConfigProvider } from '@oceanbase/design';
import type { CardProps } from '@oceanbase/design';

const CardTest: React.FC<CardProps> = props => {
  return (
    <Card title="Title" {...props}>
      Content
    </Card>
  );
};

describe('ConfigProvider card', () => {
  it('divided', () => {
    const Child1 = () => {
      const { card } = useContext(ConfigProvider.ConfigContext);
      expect(card.divided).toBe(undefined);
      return <CardTest />;
    };
    const Child2 = () => {
      const { card } = useContext(ConfigProvider.ConfigContext);
      expect(card.divided).toBe(false);
      return <CardTest />;
    };
    const Child3 = () => {
      const { card } = useContext(ConfigProvider.ConfigContext);
      expect(card.divided).toBe(false);
      return <CardTest />;
    };
    const Child4 = () => {
      const { card } = useContext(ConfigProvider.ConfigContext);
      expect(card.divided).toBe(false);
      return <CardTest divided={true} />;
    };

    const { container, asFragment } = render(
      <ConfigProvider>
        <Child1 />
        <ConfigProvider card={{ divided: false }}>
          <Child2 />
          <ConfigProvider>
            <Child3 />
            <Child4 />
          </ConfigProvider>
        </ConfigProvider>
      </ConfigProvider>
    );
    expect(container.querySelectorAll('.ant-card-no-divider').length).toBe(2);
    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
