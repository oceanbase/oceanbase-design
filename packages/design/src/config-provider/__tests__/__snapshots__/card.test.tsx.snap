// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfigProvider card > divided 1`] = `
<div
  class="ant-card ant-card-bordered ant-card-has-title"
>
  <div
    class="ant-card-head"
  >
    <div
      class="ant-card-head-wrapper"
    >
      <div
        class="ant-card-head-title"
      >
        Title
      </div>
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    Content
  </div>
</div>
`;
